### PR TITLE
[BISERVER-14644] - Users with password having certain special characters, or strings of characters, are not allowed to login to PUC

### PR DESCRIPTION
**Pull request by renato-s**
Added URL  encoding to handle special characters before b64 encoding

### CODE REVIEW

**Comment by buildguy** [comment]
### :x: Build failed in 2m 47s
Build command:
```console
mvn clean verify -B -e -Daudit -Djs.no.sandbox -pl \
extensions,user-console
```
---
 :exclamation: No tests found!
 ---
### Errors:
<details><summary><b>Filtered log</b> (click to expand)</summary>
<p>

```
script returned exit code 1
```
 
</p>
</details>
 
 

:information_source:  _This is an automatic message_

**Comment by buildguy** [comment]
<div align='center'>

[👍 Frogbot scanned this pull request and found that it did not add vulnerable dependencies.](https://docs.jfrog-applications.jfrog.io/jfrog-applications/frogbot)

</div>

<details>
<summary> <b>Note:</b> </summary>


---
<div align='center'>

**Frogbot** also supports **Contextual Analysis, Secret Detection, IaC and SAST Vulnerabilities Scanning**. This features are included as part of the [JFrog Advanced Security](https://jfrog.com/advanced-security) package, which isn't enabled on your system.

</div>


</details>


---
<div align='center'>

[🐸 JFrog Frogbot](https://docs.jfrog-applications.jfrog.io/jfrog-applications/frogbot)

</div>

File: extensions/src/main/java/org/pentaho/platform/web/http/api/resources/UserRoleDaoResource.java [hunk 1]
```diff
@@ -248,11 +250,16 @@ public Response changeUserPassword( ChangePasswordUser user ) {
     return Response.noContent().build();
   }
 
-  private String b64DecodePassword( String encodedPassword ) {
-    if ( !StringUtils.isEmpty( encodedPassword ) && encodedPassword.startsWith( "ENC:" ) ) {
-      return new String( Base64Utils.fromBase64( encodedPassword.substring( 4 ) ), StandardCharsets.UTF_8 );
-    } else {
-      return encodedPassword;
+  private String b64DecodePassword( String encodedPassword ) throws UnsupportedEncodingException {
+    try {
```
Comment (smmribeiro) [inline extensions/src/main/java/org/pentaho/platform/web/http/api/resources/UserRoleDaoResource.java:254]: Please remove the try-catch as it is only rethrowing the original exception (check SonarQube's report).

**Review by smmribeiro — CHANGES_REQUESTED** [review]

**Comment by buildguy** [comment]
<div align='center'>

[👍 Frogbot scanned this pull request and found that it did not add vulnerable dependencies.](https://docs.jfrog-applications.jfrog.io/jfrog-applications/frogbot)

</div>

<details>
<summary> <b>Note:</b> </summary>


---
<div align='center'>

**Frogbot** also supports **Contextual Analysis, Secret Detection, IaC and SAST Vulnerabilities Scanning**. This features are included as part of the [JFrog Advanced Security](https://jfrog.com/advanced-security) package, which isn't enabled on your system.

</div>


</details>


---
<div align='center'>

[🐸 JFrog Frogbot](https://docs.jfrog-applications.jfrog.io/jfrog-applications/frogbot)

</div>

**Comment by buildguy** [comment]
### :white_check_mark: Build finished in 30m 36s
Build command:
```console
mvn clean verify -B -e -Daudit -Djs.no.sandbox -pl \
extensions,user-console
```
---
:ok_hand: All tests passed!
 
Tests run: **1573**, Failures: **0**, Skipped: **5** &nbsp;&nbsp; [Test Results](https://wingman.orl.eng.hitachivantara.com/job/projects/job/platform/job/PR-5597/4/testReport)

 ---
 
 

:information_source:  _This is an automatic message_

**Review by smmribeiro — APPROVED** [review]
Looks good!

**Pull request merged**